src/common/hooks/useFacets.js

```diff
@@ -0,0 +1,223 @@
+
+const useFacets = (segmentAccordions, segmentOptions, selectedFacetFilters, getNewRecords, data) => {
+  const {
+    query: { query, filters = '' },
+    onFetchFacets,
+    parentResources: { facets },
+  } = data;
+
+  const records = facets.records[0];
+
+  const location = useLocation();
+  const [accordions, setAccordions] = useState(segmentAccordions);
+  const [accordionsData, setAccordionsData] = useState({});
+  const [facetsOptions, setFacetsOptions] = useState(segmentOptions);
+  const [facetSettings, setFacetSettings] = useState({});
+  const [facetNameToOpen, setFacetNameToOpen] = useState('');
+  const [showLoadingForAllFacets, setShowLoadingForAllFacets] = useState(false);
+
+  const prevAccordionsState = useRef(accordions);
+  const prevFilters = useRef({});
+  const prevUrl = useRef({});
+  const prevQuery = useRef('');
+
+  const onToggleSection = ({ id }) => {
+    setAccordions(curState => {
+      const newState = _.cloneDeep(curState);
+      newState[id] = !curState[id];
+      return newState;
+    });
+  };
+
+  const handleFilterSearch = (filter) => {
+    const {
+      name,
+      value,
+    } = filter;
+
+    setFacetSettings(prevFacetSettings => ({
+      ...prevFacetSettings,
+      [name]: {
+        ...prevFacetSettings[name],
+        value,
+      },
+    }));
+  };
+
+  const processFilterChange = (selectedFilters, facetName) => {
+    if (selectedFilters) {
+      const isFilterChanged = prevFilters.current[facetName]?.length !== selectedFilters.length;
+      if (isFilterChanged) {
+        prevFilters.current[facetName] = selectedFilters;
+
+        setAccordionsData(prevAccordionsData => ({
+          ...prevAccordionsData,
+          [facetName]: {
+            ...prevAccordionsData[facetName],
+            isSelected: true,
+          },
+        }));
+      }
+    } else {
+      const isLastFilterRemoved = prevFilters.current[facetName]?.length === 1 && selectedFilters === undefined;
+      if (isLastFilterRemoved) {
+        prevFilters.current[facetName] = [];
+
+        setAccordionsData(prevAccordionsData => ({
+          ...prevAccordionsData,
+          [facetName]: {
+            ...prevAccordionsData[facetName],
+            isSelected: false,
+          },
+        }));
+      }
+    }
+  };
+
+  const processOnMoreClicking = (onMoreClickedFacet) => {
+    onFetchFacets({ onMoreClickedFacet });
+
+    setFacetSettings(prevFacetSettings => ({
+      ...prevFacetSettings,
+      [onMoreClickedFacet]: {
+        ...prevFacetSettings[onMoreClickedFacet],
+        isOnMoreClicked: true,
+      },
+    }));
+  };
+
+  const processAllFacets = () => {
+    const facetsData = { ...accordionsData };
+
+    _.forEach(facetSettings, (settings, facet) => {
+      facetsData[facet] = {
+        ...facetsData[facet],
+        ...settings,
+      };
+    });
+
+    onFetchFacets({
+      accordions,
+      accordionsData: facetsData,
+    });
+  };
+
+  const handleFetchFacets = (property = {}) => {
```

Comment:
Please wrap this function in useCallback hook

Reply:
Thanks, fixed